plugin/callgraph/assignment.go

```diff
@@ -0,0 +1,80 @@
+package callgraph
+
+import (
+	"slices"
+
+	"github.com/safedep/dry/utils"
+	sitter "github.com/smacker/go-tree-sitter"
+)
+
+type assignmentNode struct {
+	Namespace  string
+	AssignedTo []string
+	TreeNode   *sitter.Node
+}
+
+func newAssignmentGraphNode(namespace string, treeNode *sitter.Node) *assignmentNode {
+	return &assignmentNode{
+		Namespace:  namespace,
+		AssignedTo: []string{},
+		TreeNode:   treeNode,
+	}
+}
+
+type AssignmentGraph struct {
+	Assignments map[string]*assignmentNode // Map of identifier to possible namespaces or other identifiers
+}
+
+func NewAssignmentGraph() *AssignmentGraph {
```

Comment:
Is it required to be exposed outside package? I think its internal to CG plugin?